server/src/main/java/io/druid/server/lookup/cache/LookupsStateWithMap.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.lookup.cache;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Same as LookupsState except that it uses LookupExtractorFactoryMapContainer instead of
```

Comment:
What's the reason for such separation?

It seems that at least at some place, you serialize LookupsStateWithMap (in LookupCoordinatorManager), and deserialize the same object as LookupsState in LookupModule.

For me it's highly counterintuitive, and complicates reading and understanding the code for future readers. When I expore inter-node connections in Druid, when I see some object is serialized and sent, I'm searching for the place when this object is accepted and processed by searching usages of the same class in the Druid codebase. If the object is deserialized as a completely different kind of object, it would be a dead-end for my search.

Also, it implicitly relies on the current serialization format, that doesn't involve the class name.